docs/install_linux.md

```diff
@@ -14,8 +14,8 @@ our release schedule.
 Install:
 
 ```bash
-curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
-echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/etc/apt/trusted.gpg.d/githubcli-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/trusted.gpg.d/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
```

Comment:
Thanks, but this change could potentially be backwards-incompatible. Is `/etc/apt/trusted.gpg.d` guaranteed to exist in all versions of Debian and Ubuntu?

Reply:
I would say at least for Debian releases since Wheezy.

In the cited documentation Debian release Wheezy was named as "most recent" version. Debian Wheezy was initially released in May 2013 ([Wheezy press release](https://www.debian.org/News/2013/20130504)).

For Debian versions before (at least since Debian Etch [released 2010](https://www.debian.org/releases/etch/)) the preferred way to import a repository key into the apt keyring was to use the tool `apt-key`.

For Debian releases before 2010 the directory `/usr/share/keyrings/` seems to be the way to go.

Reply:
I've just checked Ubuntu 20.04 and it also has both folders. It also has the `apt-key` utility that seems to import keys into `/etc/apt/trusted.gpg.d`, but in the past we've received reports that `apt-key` is “deprecated” (not sure since when or on which distros exactly) https://github.com/cli/cli/issues/2893

It looks like `/etc/apt/trusted.gpg.d` seems to be a universally preferred location, but I'm squeamish about changing the docs for everyone since up until now these instructions worked, and I'm not sure if absolutely everyone has the `/etc/apt/trusted.gpg.d` directory in their distro.

I wonder could we add a bit of shell script that writes to `/etc/apt/trusted.gpg.d` if it exists, otherwise fall back to `/usr/share/keyrings`?

Reply:
> I'm not sure if absolutely everyone has the `/etc/apt/trusted.gpg.d` directory in their distro.

The directory should be in any Debian based distro at least since 2013, the same time when `apt-key` started to become deprecated.

So the mentioned change would impact Debian based distros only.

Using a script approach seems a good solution.
- store in `/etc/apt/trusted.gpg.d` if exist
- store in `/usr/share/keyrings` if exist
- otherwise store nowhere, if both directories do not exist the distro most likely not even support signed repositories